File: src/main/java/Problem0016.java
import java.math.BigInteger;

public class Problem0016 {
    public Problem0016() {
        long startTime = System.nanoTime();
        long result = 0;

        // Solution:
        BigInteger n = new BigInteger("2");
        n = n.pow(1000);
        for (char c : n.toString().toCharArray()) {
            result += c - 48;
        }

        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + result +"\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }
}

File: src/main/cs/Problem0016.cs
using System;
using System.Diagnostics;
using System.Numerics;
using Euler.test.cs;

namespace Euler.main.cs
{
    public class Problem0016
    {
        public Problem0016()
        {
            Stopwatch stopWatch = new Stopwatch();
            stopWatch.Start();
            long result = 0;

            // Solution:
            BigInteger n = BigInteger.Pow(2, 1000);
            foreach (char c in n.ToString())
            {
                result += c - 48;
            }


            stopWatch.Stop();
            string elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + result + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                  ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                  : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                    " ms"));
            if (Test.DoBenchmark)
            {
                Benchmark.AddTime(16, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
            }
        }
    }
}